internal/services/paloalto/schema/network_profile.go

```diff
@@ -0,0 +1,332 @@
+					Computed: true,
+					Elem: &pluginsdk.Schema{
+						Type: pluginsdk.TypeString,
+					},
+				},
+
+				"egress_nat_ips": {
+					Type:     pluginsdk.TypeList,
+					Computed: true,
+					Elem: &pluginsdk.Schema{
+						Type: pluginsdk.TypeString,
+					},
+				},
+			},
+		},
+	}
+}
+
+func VHubNetworkProfileSchema() *pluginsdk.Schema {
+	return &pluginsdk.Schema{
+		Type:     pluginsdk.TypeList,
+		Required: true,
+		MaxItems: 1,
+		Elem: &pluginsdk.Resource{
+			Schema: map[string]*pluginsdk.Schema{
+				"virtual_hub_id": {
+					Type:         pluginsdk.TypeString,
+					Required:     true,
+					ForceNew:     true,
+					ValidateFunc: networkValidate.VirtualHubID,
+				},
+
+				"network_virtual_appliance_id": {
+					Type:         pluginsdk.TypeString,
+					Required:     true,
+					ForceNew:     true,
+					ValidateFunc: networkvirtualappliances.ValidateNetworkVirtualApplianceID,
+				},
+
+				"public_ip_ids": {
+					Type:     pluginsdk.TypeList,
+					Required: true,
+					MinItems: 1,
+					Elem: &pluginsdk.Schema{
+						Type:         pluginsdk.TypeString,
+						ValidateFunc: networkValidate.PublicIpAddressID,
+					},
+				},
+
+				"egress_nat_ip_ids": {
+					Type:     pluginsdk.TypeList,
+					Optional: true,
+					Elem: &pluginsdk.Schema{
+						Type:         pluginsdk.TypeString,
+						ValidateFunc: networkValidate.PublicIpAddressID,
+					},
+				},
+
+				"trusted_subnet_id": {
+					Type:     pluginsdk.TypeString,
+					Computed: true,
+				},
+
+				"untrusted_subnet_id": {
+					Type:     pluginsdk.TypeString,
+					Computed: true,
+				},
+
+				"ip_of_trust_for_udr": {
+					Type:     pluginsdk.TypeString,
+					Computed: true,
+				},
+
+				"public_ip": {
+					Type:     pluginsdk.TypeList,
+					Computed: true,
+					Elem: &pluginsdk.Schema{
+						Type: pluginsdk.TypeString,
+					},
+				},
+
+				"egress_nat_ips": {
+					Type:     pluginsdk.TypeList,
+					Computed: true,
+					Elem: &pluginsdk.Schema{
+						Type: pluginsdk.TypeString,
+					},
+				},
+			},
+		},
+	}
+}
+
+func ExpandNetworkProfileVnet(input []NetworkProfileVnet) firewalls.NetworkProfile {
+	result := firewalls.NetworkProfile{
+		EnableEgressNat: firewalls.EgressNatDISABLED,
+	}
+
+	if len(input) == 0 {
+		return result
+	}
+
+	profile := input[0]
+
+	if len(profile.PublicIPIDs) > 0 {
+		ipIDs := make([]firewalls.IPAddress, 0)
+		for _, v := range profile.PublicIPIDs {
+			ipIDs = append(ipIDs, firewalls.IPAddress{
+				ResourceId: pointer.To(v),
+			})
+		}
+		result.PublicIPs = ipIDs
+	}
+
+	if len(profile.EgressNatIPIDs) > 0 {
+		result.EnableEgressNat = firewalls.EgressNatENABLED
+		egressNatIPs := make([]firewalls.IPAddress, 0)
+		for _, v := range profile.EgressNatIPIDs {
+			egressNatIPs = append(egressNatIPs, firewalls.IPAddress{
+				ResourceId: pointer.To(v),
+			})
+		}
+		result.EgressNatIP = pointer.To(egressNatIPs)
+	}
+
+	result.NetworkType = firewalls.NetworkTypeVNET
+	vnet := profile.VnetConfiguration[0]
```

Comment:
this assumes we have a VnetConfiguration, it's not checked above?

Reply:
It's a required property and has required properties within it, so i think this is fine?